File: Definitions/Game_Definitions/classes/AttackTagHitPrereqCondition.lua
---@meta
---@diagnostic disable

---@class AttackTagHitPrereqCondition : BaseHitPrereqCondition
---@field attackTag CName
AttackTagHitPrereqCondition = {}

---@return AttackTagHitPrereqCondition
function AttackTagHitPrereqCondition.new() return end

---@param props table
---@return AttackTagHitPrereqCondition
function AttackTagHitPrereqCondition.new(props) return end

---@param hitEvent gameeventsHitEvent
---@return Bool
function AttackTagHitPrereqCondition:Evaluate(hitEvent) return end

---@param recordID TweakDBID|string
function AttackTagHitPrereqCondition:SetData(recordID) return end

File: DecompiledGameScripts/core/gameplay/prereqs/damage/attackTagHitPrereqCondition.swift

public class AttackTagHitPrereqCondition extends BaseHitPrereqCondition {

  public let m_attackTag: CName;

  public func SetData(recordID: TweakDBID) -> Void {
    this.m_attackTag = TweakDBInterface.GetCName(recordID + t".attackTag", n"None");
    super.SetData(recordID);
  }

  public func Evaluate(hitEvent: ref<gameHitEvent>) -> Bool {
    let attackRecord: ref<Attack_GameEffect_Record> = hitEvent.attackData.GetAttackDefinition().GetRecord() as Attack_GameEffect_Record;
    let result: Bool = Equals(attackRecord.AttackTag(), this.m_attackTag);
    return this.m_invert ? !result : result;
  }
}
